services/QuillLMS/app/workers/set_impact_metrics_worker.rb

```diff
@@ -3,33 +3,30 @@
 class SetImpactMetricsWorker
   include Sidekiq::Worker
 
-  ACTIVITY_SESSION_MINIMUM = 9
   FREE_LUNCH_MINIMUM = 39
 
   def perform
     set_impact_metrics
   end
 
   def set_impact_metrics
-    finished_activity_sessions_query = ActivitySession.unscoped.where(state: 'finished')
-    teachers_query = User.select(:id)
-      .joins(:units, classroom_units: :activity_sessions)
-      .where("activity_sessions.state = 'finished'")
-      .group('users.id')
-      .having('count(activity_sessions) > ?', ACTIVITY_SESSION_MINIMUM)
-    teacher_ids = teachers_query.to_a.map(&:id)
-    schools_query = School.joins(:schools_users).where(schools_users: {user_id: teacher_ids}).distinct
-    low_income_schools_query = schools_query.where("free_lunches > ?", FREE_LUNCH_MINIMUM)
+    finished_activity_sessions_query = QuillBigQuery::ActivitiesAllTimeQuery.run
+
+    teachers_query = QuillBigQuery::ActiveTeachersAllTimeQuery.run
+    teacher_ids = teachers_query.to_a.map {|teacher| teacher["id"]}
+
+    schools_query = QuillBigQuery::SchoolsContainingCertainTeachersQuery.run(teacher_ids)
+    low_income_schools_query = schools_query.select { |school| school["free_lunches"].present? && school["free_lunches"] > FREE_LUNCH_MINIMUM}
 
     number_of_sentences = finished_activity_sessions_query.count.floor(-5) * 10
```

Comment:
(Also, I realize that this isn't actually your code, but I believe you can remove all references to `floor` in these calculations since we expect all of our counts to be integers, and `floor` is used to is used to figure out how many decimal places to show for a floating point number: https://apidock.com/ruby/Numeric/floor.)

Reply:
The "floor" is needed not to round to an integer, but because we are rounding these numbers down to the nearest 10,000 (because these are very big numbers so it makes for better display to show them nicely rounded down to five zeros).